include/zephyr/senss/senss_datatypes.h

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef SENSS_DATATYPES_H_
+#define SENSS_DATATYPES_H_
+
+/**
+ * @brief Data Types
+ * @addtogroup senss_datatypes
+ * @{
+ */
+
+/**
+ * @def SENSS_SENSOR_FLAG_PHYSICAL
+ * @brief physical sensor flag
+ */
+#define SENSS_SENSOR_FLAG_PHYSICAL BIT(0);
+
+/**
+ * @def SENSS_SENSOR_FLAG_VIRTUAL
+ * @brief virtual sensor flag
+ */
+#define SENSS_SENSOR_FLAG_VIRTUAL  BIT(1);
+
+
+/**
+ * @struct senss_sensor_value_header
+ * @brief sensor value header
+ *
+ * Each sensor value data structure should have this header
+ *
+ * Here use 'base_timestamp' (uint64_t) and 'timestampe_delta' (uint32_t) to
+ * save memory usage in batching mode.
+ *
+ * The 'base_timestamp' is for readings[0], the 'timestamp_delta' is relation
+ * to the previous 'readings'. So,
+ * timestamp of readings[0] is
+ *	header.base_timestamp + readings[0].timestamp_delta.
+ * timestamp of readings[1] is
+ *	timestamp of readings[0] + readings[1].timestamp_delta.
+ *
+ * Since timestamp unit is nano seconds, the max 'timestamp_delta' (uint32_t)
+ * is 4.295 seconds.
+ *
+ * If a sensor has batched data where two consecutive readings differ by
+ * more than 4.295 seconds, the sensor subsystem core will split them
+ * across multiple instances of the readings structure, and send multiple
+ * events.
+ *
+ * This concept is borrowed from CHRE:
+ * https://cs.android.com/android/platform/superproject/+/master:\
+ * system/chre/chre_api/include/chre_api/chre/sensor_types.h
+ */
+struct senss_sensor_value_header {
+	/** base timestamp of this data readings, unit is nano seconds */
+	uint64_t base_timestamp;
+	/** count of this data readings */
+	uint16_t reading_count;
+};
+
+/**
+ * Sensor value data structure types based on common data types.
+ * suitable for common sensors, such as IMU, Light sensors
+ */
+
+struct senss_sensor_value_3d_double {
+	struct senss_sensor_value_header header;
+	struct {
+		/** relation timestamp to the previous readings,
+		 * unit is nano seconds */
+		uint32_t timestamp_delta;
```

Comment:
Changed time stamp unit from nano seconds to micro seconds, since micro seconds is enough. 
Then the roll-over time is 4925 seconds,  so its not easy for roll-over. Even roll-over with very very slow sensors (interval exceeds 4925 seconds), we can split to multiple data report structures.

Reply:
Some windows sensors, such as activity (running/walking/...), pedometer support history mode, which requires the sensor hub to batch/buffer of the sensor data as much as possible, this can save more power, since when system is in connected standby, the sensor hub can still working and buffering the data as much as possible. 

The sample rate of activity and pedometer sensor in history mode is typically 1 minute. If using nano seconds, with uint32_t delta, it will always overflow. 

Some HW can have very high accuracy clock, such as 15MHz+, even just 1MHz (micro seconds), seems most sensor usages can't leverage so high accuracy timing, if there have any examples, please tell me. 

So, I think micro second unit is more better.

Reply:
Ack, thanks